### PR TITLE
mezo added to trainer_builder

### PR DESCRIPTION
**Pull request by thistleknot**
added mezo

### CODE REVIEW

**Comment by thistleknot** [comment]
added print statements to see the outputted train loss
image
not sure if you know what needs to be done to ensure the old behaviour

**Comment by thistleknot** [comment]
hmmm, 3 epochs and training loss failed to show evidence of improvement

image

using
```
base_model: /home/user/text-generation-webui/models/open_llama_3b_v2/
base_model_config: /home/user/text-generation-webui/models/open_llama_3b_v2/
model_type: LlamaForCausalLM
tokenizer_type: LlamaTokenizer
load_in_8bit: false
load_in_4bit: true
strict: false
datasets:
#  - path: /home/user/datasets/mine/dolly_150.jsonl
#    type: alpaca
  - path: /home/user/datasets/mine/orca_100.jsonl
    type:
      format: "[INST]System:\n\n{input}\n\nInstruction:\n\n{instruction}\n\nAnswer:\n\n[/INST]"
      no_input_format: "[INST]Instruction:\n\n{instruction}\n\nAnswer:\n\n[/INST]"
  - path: /home/user/datasets/mine/Philosophy.jsonl
    type: completion
  - path: /home/user/datasets/mine/spo.jsonl
    type:
      system_prompt: ""
      field_system: system
      format: "[INST]Context:{input}\n\nInstruction:\n\n{instruction}\n\nAnswer:\n\n[/INST]"
  - path: /home/user/datasets/mine/aa.jsonl
    type:
      system_prompt: ""
      field_system: system
      format: "[INST]Context:{input}\n\nInstruction:\n\n{instruction}\n\nAnswer:\n\n[/INST]"
dataset_prepared_path:
val_set_size: 0.01
adapter: qlora
lora_model_dir:
sequence_len: 2048
sample_packing: true
lora_r: 8
lora_alpha: 32
lora_dropout: 0.05
lora_target_modules:
lora_target_linear: true
lora_fan_in_fan_out:
wandb_project:
wandb_entity:
wandb_watch:
wandb_run_id:
wandb_log_model:
output_dir: ./qlora-out
gradient_accumulation_steps: 1
micro_batch_size: 3
num_epochs: 3
optimizer: adamw_bnb_8bit
torchdistx_path:
lr_scheduler: cosine
learning_rate: 0.001
train_on_inputs: false
group_by_length: false
bf16: false
fp16: true
tf32: false
gradient_checkpointing: true
early_stopping_patience: 
resume_from_checkpoint:
local_rank:
logging_steps: 1
xformers_attention:
flash_attention: false
gptq_groupsize:
gptq_model_v1:
warmup_steps: 5
eval_steps: 25
save_steps: 25
max_steps: 1000
debug:
deepspeed:
weight_decay: 0.1
fsdp:
fsdp_config:
special_tokens:
  bos_token: "<s>"
  eos_token: "</s>"
  unk_token: "<unk>"

```
accelerate launch scripts/finetune.py examples/openllama-3b/qlora-mine.yml
